File: Coding Ninjas - CodeStudio/Basics Of Javascript Guided Path/sort012.js
function sort012(arr, n) {
    let low = 0, mid = 0, high = n - 1;
    while(mid <= high) {
        if(arr[mid] == 0) {
            let temp = arr[mid];
            arr[mid] = arr[low];
            arr[low] = temp;
            low++;
            mid++;
        }
        else if(arr[mid] == 2) {
            let temp = arr[mid];
            arr[mid] = arr[high];
            arr[high] = temp;
            high--;
        }
        else {
            mid++;
        }
    }
}

File: Coding Ninjas - CodeStudio/Basics of Java Guided Path/sort012.java
public class Solution 
{
    public static void sort012(int[] arr)
    {
        int l = 0, m = 0, r = arr.length - 1;
        while(m <= r) {
            if(arr[m] == 0) {
                int temp = arr[l];
                arr[l] = arr[m];
                arr[m] = temp;
                l++;
                m++;
            }
            else if(arr[m] == 2) {
                int temp = arr[m];
                arr[m] = arr[r];
                arr[r] = temp;
                r--;
            }
            else {
                m++;
            }
        }
    }
}